script/playbackstrategy/modifiers/live/restartable.js

```diff
@@ -1,87 +1,71 @@
         var self = this;
 
-        function determineTimeUntilStartOfWindow () {
-          mediaPlayer.addEventCallback(self, detectCurrentTimeCallback);
-        }
-
-        function stopDeterminingTimeUntilStartOfWindow () {
-          mediaPlayer.removeEventCallback(self, detectCurrentTimeCallback);
-        }
+        var callbacksMap = [];
+        var startTime = Date.now();
+        var fakeTimer = {};
 
-        function detectCurrentTimeCallback (event) {
-          if (event.state === MediaPlayerBase.STATE.PLAYING && event.currentTime > 0) {
-            removeEventCallback(self, detectCurrentTimeCallback);
-            millisecondsUntilStartOfWindow = event.currentTime * 1000;
-            determineTimeSpentBuffering();
+        function updateFakeTimer (state) {
+          if (fakeTimer.wasPlaying && fakeTimer.runningTime) {
+            fakeTimer.currentTime += (Date.now() - fakeTimer.runningTime) / 1000;
           }
+          fakeTimer.runningTime = Date.now();
+          fakeTimer.wasPlaying = state === MediaPlayerBase.STATE.PLAYING;
         }
 
         function autoResumeAtStartOfRange () {
-          if (millisecondsUntilStartOfWindow !== null) {
-            var resumeTimeOut = Math.max(0, millisecondsUntilStartOfWindow - AUTO_RESUME_WINDOW_START_CUSHION_MILLISECONDS);
-            var pauseStarted = new Date().getTime();
-            var autoResumeTimer = setTimeout(function () {
-              removeEventCallback(self, detectIfUnpaused);
-              millisecondsUntilStartOfWindow = 0;
-              resume();
-            }, resumeTimeOut);
+          var resumeTimeOut = Math.max(0, getCurrentTime() - getSeekableRange().start - AUTO_RESUME_WINDOW_START_CUSHION_SECONDS);
+          var autoResumeTimer = setTimeout(function () {
+            removeEventCallback(self, detectIfUnpaused);
+            resume();
+          }, resumeTimeOut * 1000);
 
-            addEventCallback(self, detectIfUnpaused);
-          }
+          addEventCallback(self, detectIfUnpaused);
 
           function detectIfUnpaused (event) {
             if (event.state !== MediaPlayerBase.STATE.PAUSED) {
               removeEventCallback(self, detectIfUnpaused);
               clearTimeout(autoResumeTimer);
-              var timePaused = new Date().getTime() - pauseStarted;
-              millisecondsUntilStartOfWindow -= timePaused;
             }
           }
         }
 
         function addEventCallback (thisArg, callback) {
-          mediaPlayer.addEventCallback(thisArg, callback);
-        }
+          function newCallback (event) {
+            updateFakeTimer(event.state);
 
-        function removeEventCallback (thisArg, callback) {
-          mediaPlayer.removeEventCallback(thisArg, callback);
+            event.currentTime = getCurrentTime();
+            event.seekableRange = getSeekableRange();
+            callback(event);
+          }
+          callbacksMap.push({ from: callback, to: newCallback });
```

Comment:
If/when we fix the problem with the underlying player's event callbacks, we might want to extract this functionality and re-use. Should probably raise an issue for that actually...